File: python/tests/model/identity_test.py
from typing import Any, Dict

import pytest
from openral_py.model.identity import Identity


class TestIdentity:
    def test_from_map_with_empty_data(self):
        # arrange
        data = {
            "UID": "123",
        }

        # act
        identity = Identity.from_map(data)

        # assert
        assert identity.uid == "123"
        assert identity.name is None
        assert identity.site_tag is None
        assert identity.alternate_ids == []
        assert identity.alternate_names == []

    def test_from_map_with_full_data(self):
        # arrange
        data : Dict[str, Any] = {
            "UID": 123,
            "name": "John",
            "siteTag": "example.com",
            "alternateIDs": [456, 789],
            "alternateNames": ["Johnny"],
        }

        # act
        identity = Identity.from_map(data)

        # assert
        assert identity.uid == 123
        assert identity.name == "John"
        assert identity.site_tag == "example.com"
        assert identity.alternate_ids == [456, 789]
        assert identity.alternate_names == ["Johnny"]

File: dart/test/model/identity_test.dart
import 'package:openral_core/src/model/identity.dart';
import 'package:test/test.dart';

void main() {
  group(
    Identity,
    () {
      group('toMap', () {
        test('returns a map with values', () async {
          //arrange
          final identity = Identity(
            uid: "uid",
            name: "name",
            siteTag: "siteTag",
            alternateIDs: ["alternateIDs"],
            alternateNames: ["alternateNames"],
          );

          //act
          final map = identity.toMap();

          //assert
          expect(
            map,
            equals({
              "UID": "uid",
              "name": "name",
              "siteTag": "siteTag",
              "alternateIDs": ["alternateIDs"],
              "alternateNames": ["alternateNames"]
            }),
          );
        });

        test('if values are null, return a map with null values', () async {
          //arrange
          final identity = Identity(
            uid: "uid",
            name: null,
            siteTag: null,
            alternateIDs: [],
            alternateNames: [],
          );

          //act
          final map = identity.toMap();

          //assert

          expect(
            map,
            equals({
              "UID": "uid",
              "name": null,
              "siteTag": null,
              "alternateIDs": [],
              "alternateNames": [],
            }),
          );
        });
      });
    },
  );
}
